OMEdit/OMEditLIB/Element/Element.cpp

```diff
@@ -554,8 +554,10 @@ Element::Element(ModelInstance::Component *pModelComponent, bool inherited, Grap
     ModelInstance::CoordinateSystem coordinateSystem = getCoOrdinateSystemNew();
     qreal initialScale = coordinateSystem.getInitialScale();
     QVector<QPointF> extent;
-    extent.append(QPointF(initialScale * boundingRect().left(), initialScale * boundingRect().top()));
-    extent.append(QPointF(initialScale * boundingRect().right(), initialScale * boundingRect().bottom()));
+    qreal xExtent = initialScale * boundingRect().width() / 2;
+    qreal yExtent = initialScale * boundingRect().height() / 2;
+    extent.append(QPointF(-xExtent, -yExtent));
+    extent.append(QPointF(xExtent, yExtent));
```

Comment:
This seems to be doing the same thing. We might need to consider the case when either width or height is zero.

Reply:
I try this code with models with {{0,0},{0,0}} icon and diagram extents and it works fine